tests/handlers/records-query.spec.ts

```diff
@@ -202,6 +202,62 @@ export function testRecordsQueryHandler(): void {
         expect(reply3.entries?.length).to.equal(0);
       });
 
+      it('should be able to query for published records', async () => {
```

Comment:
Please add tests for getting only unpublished records. For authorized and anonymous/unauthorized access.

Daniel is interested in the authorized case, I am more interested in the later as it would be a security breach.

Reply:
@thehenrytsai I added the additional test cases and logic, I think they're covering the various edge cases that exist.

I also added some schema validation that will disallow `published` to be set to false as well as a `datePublished` range to be specified within a filter.